crates/energy_optimization/src/current_minimizer.rs

```diff
@@ -0,0 +1,87 @@
+use std::f32::EPSILON;
+
+use serde::{Deserialize, Serialize};
+use serialize_hierarchy::SerializeHierarchy;
+use types::{
+    joints::Joints, motor_commands::MotorCommands, parameters::CurrentMinimizerParameters,
+};
+
+#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize, SerializeHierarchy)]
+enum State {
+    #[default]
+    Optimizing,
+    Resetting,
+}
+
+#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize, SerializeHierarchy)]
+pub struct CurrentMinimizer {
+    position_offset: Joints<f32>,
+    state: State,
+    last_motor_commands: MotorCommands<Joints<f32>>,
+    last_positions: Joints<f32>,
+    parameters: CurrentMinimizerParameters,
+}
+
+impl CurrentMinimizer {
+    pub fn new() -> Self {
+        Self::default()
+    }
```

Comment:
Drop this method. This is only calling `Default`, thus call `Default` at the call side

Reply:
Done